### ISSUE TITLE
Volcano Scheduler Memory leak

### ISSUE THREAD

**Issue by tianshimoyi** [volcano-sh/volcano#3205]
**What happened**:

Check the memory usage of volcano scheduler. It is consistent with the increase. After restarting, the memory plummets.

### Monitor before restarting

<img width="711" alt="截屏2023-11-16 10 26 47" src="https://github.com/volcano-sh/volcano/assets/68532999/b180066e-1be7-491d-b0cb-f18e40d1ec36">

### Monitor after restart

<img width="697" alt="截屏2023-11-16 10 29 27" src="https://github.com/volcano-sh/volcano/assets/68532999/5e3062e9-5a27-4bd8-b465-a5022bec6c0b">

### Heap memory usage couple before restart
![截屏2023-11-16 10 30 35](https://github.com/volcano-sh/volcano/assets/68532999/fb8fde22-9fce-4242-9c54-6a7f5fce69fe)

### Heap memory usage couple after reboot
![截屏2023-11-16 10 31 15](https://github.com/volcano-sh/volcano/assets/68532999/bac05eab-01a0-44e3-ad1b-a8c4adc55b0f)

**Comment by Monokaix** [comment]
Hi, have you set memory limit of container resource limit and what's the log level you set?

**Comment by tianshimoyi** [comment]
@Monokaix After modifying the log level, the memory is still growing. Log printing speed is not fast either
![截屏2023-11-16 21 17 28](https://github.com/volcano-sh/volcano/assets/68532999/258f5b9f-626d-4013-8f95-c473d6e2f59f)
![截屏2023-11-16 21 18 30](https://github.com/volcano-sh/volcano/assets/68532999/89d22ee1-0837-4498-8927-a60d0c88a756)

**Comment by Monokaix** [comment]
Can you wait memory hit the limit and check whether memory reclamation happened？

**Comment by tianshimoyi** [comment]
@Monokaix I'll give it a try. I found a problem. The scheduler has the nodeSelector option, but the pods do not consider filtering out the situations that are not on these nodes, causing the memory to maintain the pod information on all nodes in the entire cluster, resulting in a waste of memory.

**Comment by Monokaix** [comment]
> @Monokaix I'll give it a try. I found a problem. The scheduler has the nodeSelector option, but the pods do not consider filtering out the situations that are not on these nodes, causing the memory to maintain the pod information on all nodes in the entire cluster, resulting in a waste of memory.

That's right, but I think it's not the main cause of high memory usage: ).
Much log and delayed memory reclamation may cause this phenomenon and memory can be reclaimed when it about ro hit the memory limit.

**Comment by tianshimoyi** [comment]
@Monokaix Thank you very much. I will lower the limit and observe it for a while.

**Comment by tianshimoyi** [comment]
@Monokaix I reduced the memory limit, but oom did not trigger gc, and the memory continued to increase.
![截屏2023-11-17 18 16 56](https://github.com/volcano-sh/volcano/assets/68532999/90e7dccd-b488-4351-8bb4-ecce8497918f)

<img width="626" alt="截屏2023-11-17 18 17 23" src="https://github.com/volcano-sh/volcano/assets/68532999/d4fab22d-d129-43c4-978e-b33817205548">

**Comment by Monokaix** [comment]
You can use kill -12 $volcano-scheduler pid to dump cache info.

**Comment by Monokaix** [comment]
> @Monokaix Thank you very much. I will lower the limit and observe it for a while.

What's the resources num of you cluster? like nodes and pods, we should not lower the memory too low, and meet the basic memory needs first, then lower the memory limit a bit.

**Comment by Monokaix** [comment]
Can this https://github.com/volcano-sh/volcano/pull/3435 fixed your problem?

**Comment by JesseStutler** [comment]
Hi @tianshimoyi . Have we solved this issue?
